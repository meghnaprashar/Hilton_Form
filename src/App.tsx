import React, { StatelessComponent } from "react";
import styled from "./styled-components";
import logo from "./logo.svg";
import RoomRequestForm from "./RoomRequestForm";

const Section = styled("section")`
  width: 80vw;
  min-height: 100vh;
  margin: 20px auto 40px auto;
  text-align: center;
`;

const Header = styled("header")`
  margin: 20px 0;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-start;
  font-size: calc(10px + 2vmin);
  color: black;
`;

const Logo = styled("img")`
  height: 40vmin;
`;

const FormWrapper = styled("main")`
  width: 800px;
  margin: 0 auto;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
  align-items: flex-start;
`;

const App: StatelessComponent = () => (
  <Section>
    <Header role="banner">
      <Logo src={logo} />
      <h1>Hilton Assessment 2</h1>
    </Header>
    <FormWrapper role="main">
      <RoomRequestForm />
    </FormWrapper>
  </Section>
);

export default App;
