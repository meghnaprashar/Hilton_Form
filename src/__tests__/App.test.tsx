import React from "react";
import { render } from "react-testing-library";
import App from "../App";

describe("<App />: structure", () => {
  it("renders <App/> without crashing", () => {
    const { container } = render(<App />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
