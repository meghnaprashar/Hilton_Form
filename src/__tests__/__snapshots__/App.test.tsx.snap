// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App />: structure renders <App/> without crashing 1`] = `
.c0 {
  width: 80vw;
  min-height: 100vh;
  margin: 20px auto 40px auto;
  text-align: center;
}

.c1 {
  margin: 20px 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  font-size: calc(10px + 2vmin);
  color: black;
}

.c2 {
  height: 40vmin;
}

.c3 {
  width: 800px;
  margin: 0 auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

<section
  class="c0"
>
  <header
    class="c1"
    role="banner"
  >
    <img
      class="c2"
      src="logo.svg"
    />
    <h1>
      Hilton Assessment 2
    </h1>
  </header>
  <main
    class="c3"
    role="main"
  />
</section>
`;
